src/main/java/net/snowflake/ingest/streaming/internal/FileColumnProperties.java

```diff
@@ -93,6 +100,16 @@ public void setColumnOrdinal(int columnOrdinal) {
     this.columnOrdinal = columnOrdinal;
   }
 
+  @JsonProperty("fieldId")
+  @JsonInclude(JsonInclude.Include.NON_DEFAULT)
```

Comment:
The server validates incoming requests and will expect fieldId to be populated for iceberg tables, even if its 0.
By using NON_DEFAULT we end up seeing a weird behavior where the server cannot differentiate between "null/0 because client intended to not set the value versus null/0 because of the NON_DEFAULT attribute".

Please use NON_NULL and make the field an Integer (so its nullable).
If both RowBufferStats.fieldId and FileColumnProperties.fieldId are nullable, line 55 here can stay as-is. We'll add a conditional to where RowBufferStats is initialized and only pass in a fieldId value for iceberg tables. This way null implies not-set, and not-serialized, whereas non-null implies it was explicitly set.

Reply:
Changed to `NON_NULL` and Integer.